oop2-encapsulate-inherit.js

```diff
@@ -1,41 +1,270 @@
+//     //Create methods that return the values of these private variables in case other pieces of the code need to access the data of the 4 classes.
+//     //Also create methods that allow outside code to "set" new values for the properties on the classes, without directly manipulating the values themselves.
 
+//     //your code here...
 
+class Book {
+    #author;
+    #publisher;
+    constructor() {
+        this.#author = null;
+        this.#publisher = null;
+    }
+    getAuthor() {
+        return this.#author;
+    }
+    setAuthor(author) {
+        this.#author = author;
+    }
+    getPublisher() {
+        return this.#publisher;
+    }
+    setPublisher(publisher) {
+        this.#publisher = publisher;
+    }
 
+}
 
+class Author {
+    #name;
+    #books;
+    constructor() {
+        this.#name = null;
+        this.#books = null;
+    }
+    getAuthorName() {
+        return this.#name;
+    }
+    setAuthorName(authorName) {
+        this.#name = authorName;
+    }
+    getBooks() {
+        return this.#books;
+    }
+    setBooks(books) {
+        this.#books = books;
+    }
 
+}
 
+class Publisher {
+    #authors;
+    #books;
+    constructor() {
+        this.#authors = null;
+        this.#books = null;
+    }
+    getAuthors() {
+        return this.#authors;
+    }
+    setAuthors(authors) {
+        this.#authors = authors;
+    }
+    getBooks() {
+        return this.#books;
+    }
+    setBooks(books) {
+        this.#books = books;
+    }
 
-2. Create the following classes:
-    - Umbrella class.  It should have a an organization name.  It should only have a single instance.  Umbrella is the "Parent Organization" e.g. Allegis Group
-    - Company class. It should have a compnay name and # of employees.  Create at least 3 different companies that are children of the Umbrella Organization.  (e.g. TEKsystems, Aerotek, etc.)
-    - Site class. It should have the name of the company, and the location of the site.  It is a child class that inherits from the Company class.  Create 3 sites for each company.
-    - Employee class. It should have a employee name, job title, and salary properties. Create 10 employee instances: 1 CEO, 1 Manager, 1 Secretary, 2 Engineers, 1 Financial Officer, 1 Janitor, 1 Marketer, 1 HR Personnel, 1 Lawyer.  Feel free to add other employee roles at your own discretion.
-    
-    - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
+}
+class Review {
+    #review;
+    #user;
+    constructor() {
+        this.#review = null;
+        this.#user = null;
+    }
+    getReview() {
+        return this.#review;
+    }
+    setReview(review) {
+        this.#review = review;
+    }
+    getUser() {
+        return this.#user;
+    }
+    setUser(user) {
+        this.#user = user;
+    }
 
+}
 
```

Comment:
Each of your classes need a constructor that takes parameters. Right now the only way to create anything is to create null versions and use the setters. But if you use a constructor, you can take data in the form of Book, Author, Review, and Publisher objects and use the setters and getters to change the data.